pylint_nautobot/replaced_models.py

```diff
@@ -61,3 +74,77 @@ def visit_importfrom(self, node):
                     self.add_message("nb-replaced-ipam-role", node=node)
                 elif name == "Aggregate":
                     self.add_message("nb-replaced-aggregate", node=node)
+
+class NautobotReplacedModelsImportChecker(BaseChecker):
+    """Visit 'import from' statements to find usage of models that have been replaced in 2.0."""
+
+    __implements__ = IAstroidChecker
+
+    version_specifier = ">=2,<3"
+
+    name = "nautobot-replaced-models-relationship-fields"
+    msgs = {
+        "E4311": (
+            "Relationship field is related to a model that has been replaced (%s -> %s).",
+            "nb-replaced-model-relationship-field",
```

Comment:
Possibly to reduce confusion with the Custom Relationships feature, replace `relationship` with `relation` or `related object` throughout?